test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -133,9 +133,7 @@ public void tearDown() throws Exception {
         ensureNoInitializingShards();
         ensureAllFreeContextActionsAreConsumed();
 
-        SearchService searchService = getInstanceFromNode(SearchService.class);
-        assertThat(searchService.getActiveContexts(), equalTo(0));
-        assertThat(searchService.getOpenScrollContexts(), equalTo(0));
+        ensureAllContextsReleased(getInstanceFromNode(SearchService.class));
```

Comment:
I know people argued this would increase test runtimes, but not really in my opinion. This is failing at a low rate and we only really add a delay to those tests where the assertions fails, so not a big deal IMO.

Reply:
FWIW I was one of those people, and I still don't like this change. Fire-and-forget actions on the `GENERIC` pool is a bad code smell IMO, it's going to cause us problems at scale at some point. I'm not going to ask for rework here, just recording that opinion.

Reply:
The scale problem started worrying me as well, opened https://github.com/elastic/elasticsearch/pull/111156 right after :D